src/components/dropdown/react/Dropdown.tsx

```diff
@@ -64,12 +85,63 @@ const DEFAULT_PROPS: IDefaultPropsType = {};
 const Dropdown: React.FC<DropdownProps> = ({
     children,
     className = '',
-    ...props
-}: DropdownProps): React.ReactElement => (
-    <div className={classNames(className, handleBasicClasses({ prefix: CLASSNAME }))} {...props}>
-        {children}
-    </div>
-);
+    closeOnClick = DEFAULT_PROPS.closeOnClick,
+    escapeClose = DEFAULT_PROPS.escapeClose,
+    offset,
+    // overToggle = DEFAULT_PROPS.overToggle,
+    position,
+    toggleElement,
+    width,
+}: DropdownProps): React.ReactElement => {
+    const wrapperRef: React.RefObject<HTMLDivElement> = useRef(null);
+    const [isOpen, setIsOpen]: [boolean, (isOpen: boolean) => void] = useState(Boolean(false));
+
+    function closeDropdown(): void {
+        setIsOpen(false);
+    }
+
+    function toggleDropdown(): void {
+        setIsOpen(!isOpen);
+    }
+
+    const anchorElement: React.ReactNode = (
+        <div className={`${CLASSNAME}__toggle`} onClick={toggleDropdown}>
+            {toggleElement}
+        </div>
+    );
+
+    const popperElement: React.ReactNode = isOpen && (
+        <div className={`${CLASSNAME}__menu`} style={{ width }}>
+            <div className={`${CLASSNAME}__content`}>{children(setIsOpen)}</div>
+        </div>
+    );
+
+    // Any click away from the dropdown container will close it.
+    useClickAway(wrapperRef, () => {
+        if (!closeOnClick) {
+            return;
+        }
+        closeDropdown();
+    });
+
+    return (
+        <div
+            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME }), {
+                [`${CLASSNAME}--has-toggle`]: toggleElement,
```

Comment:
You could use `hasToggle: toggleElement` instead, just after prefix.

Reply:
👍 yup I need to refactor this now that you've tweaked the helper